subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/ContentFilteringRepository.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.repositories;
+
+import org.gradle.api.Action;
+
+import javax.annotation.Nullable;
+
+public interface ContentFilteringRepository {
+
+    @Nullable
```

Comment:
We do some dancing around the null for the missing content filter (e.g. `DefaultMavenRepositoryContentDescriptor`), was wondering if we should instead use `Actions.doNothing`. Not sure how "bad" it would be as it would invalidate the shortcut in `FilteredModuleComponentRepository`:55

Reply:
Good point, let me attempt that clean up.
The test can then become a `DO_NOTHING` equality test.